### PR TITLE
ARROW-3449: [C++] Fixes to build with CMake 3.2. Document what requires newer CMake

### PR DESCRIPTION
**Pull request by wesm**
This also resolves ARROW-3984

### CODE REVIEW

File: cpp/src/plasma/CMakeLists.txt [hunk 1]
```diff
@@ -20,16 +20,12 @@ add_custom_target(plasma)
...
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/../python/cmake_modules")
-
-find_package(PythonLibsNew REQUIRED)
 find_package(Threads)
 
 # The SO version is also the ABI version
 set(PLASMA_SO_VERSION "${ARROW_SO_VERSION}")
 set(PLASMA_FULL_SO_VERSION "${ARROW_FULL_SO_VERSION}")
 
-include_directories(SYSTEM ${PYTHON_INCLUDE_DIRS})
```
Comment (wesm) [inline cpp/src/plasma/CMakeLists.txt]: I'm not sure what this Python cruft was doing here, but it is not needed for Plasma

**Comment by wesm** [comment]
BTW CMake 3.2 is available in conda with `conda install cmake=3.2`. Made setting up a separate build toolchain pretty simple

**Review by kszucs — APPROVED** [review]
LGTM

**Comment by kszucs** [comment]
Now We have a nightly test for building with CMake 3.2

**Comment by wesm** [comment]
That's great, thank you

**Pull request closed without merging**